src/utils.js

```diff
@@ -60,15 +60,21 @@ function getLocFromIndex(sourceCode, index) {
 
 // Copied directly from Relay
 function getModuleName(filePath) {
-  const filename = path.basename(filePath, path.extname(filePath));
+  // index.js -> index
+  // index.js.flow -> index.js
+  let filename = path.basename(filePath, path.extname(filePath));
+
+  // index.js -> index (when extension has multiple segments)
+  // index.react -> index (when extension has multiple segments)
+  filename = filename.replace(
+    /(?:\.js|\.react|\.react-test|\.example|\.jsx?|\.tsx?|\.flow|\.brands)+$/,
```

Comment:
do we need .hooks somewhere in here?

Reply:
I only added brands because it's a super common extension (unfortunately), but there's only a handful of `.hooks.js` files. They just need to include the Hooks suffix (`Test.hooks.js`  requires the `TestHooks` prefix.